aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/DomainType.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 Amazon.com,
+ * Inc. or its affiliates. All Rights Reserved.
+ * <p>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazonaws.mobileconnectors.appsync;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The type of domain specified in API endpoint.
+ */
+enum DomainType {
+    /**
+     * Standard domain type composed of AWS AppSync endpoint where unique-id is made of 26 alphanumeric characters.
+     * See <a href="https://docs.aws.amazon.com/general/latest/gr/appsync.html">AppSync Endpoints</a>
+     */
+    STANDARD,
+
+    /**
+     * Custom domain defined by the user.
+     */
+    CUSTOM;
+
+    private static final String STANDARD_ENDPOINT_REGEX =
+            "^https://\\w{26}\\.appsync-api\\.\\w{2}(?:(?:-\\w{2,})+)-\\d\\.amazonaws.com/graphql$";
```

Comment:
Shouldn't `/` be escaped?

Reply:
Also why `\` in `\w` needs to be escaped?

Reply:
> Shouldn't `/` be escaped?

No, as this is being used for pattern matcher. Here, we are literally looking to match `//` and hence escaping it would be redundant.

> Also why `\` in `\w` needs to be escaped?
This portion is regex expression and hence we need double backslashes to protect them from being interpreted as escape sequence from the compiler. (`\\w` will be transformed to `\w` which is valid regex construct)